libraries/chain/asset_evaluator.cpp

```diff
@@ -56,33 +56,40 @@ void_result asset_create_evaluator::do_evaluate( const asset_create_operation& o
-   {
-      auto dotpos = op.symbol.find( '.' );
-      if( dotpos != std::string::npos )
+      if( d.head_block_time() <= HARDFORK_409_TIME )
       {
-         auto prefix = op.symbol.substr( 0, dotpos );
-         auto asset_symbol_itr = asset_indx.find( op.symbol );
-         FC_ASSERT( asset_symbol_itr != asset_indx.end(), "Asset ${s} may only be created by issuer of ${p}, but ${p} has not been registered",
-                    ("s",op.symbol)("p",prefix) );
-         FC_ASSERT( asset_symbol_itr->issuer == op.issuer, "Asset ${s} may only be created by issuer of ${p}, ${i}",
-                    ("s",op.symbol)("p",prefix)("i", op.issuer(d).name) );
+         auto dotpos = op.symbol.find( '.' );
+         if( dotpos != std::string::npos )
+         {
+            auto prefix = op.symbol.substr( 0, dotpos );
+            auto asset_symbol_itr = asset_indx.find( op.symbol );
+            FC_ASSERT( asset_symbol_itr != asset_indx.end(), "Asset ${s} may only be created by issuer of ${p}, but ${p} has not been registered",
+                       ("s",op.symbol)("p",prefix) );
+            FC_ASSERT( asset_symbol_itr->issuer == op.issuer, "Asset ${s} may only be created by issuer of ${p}, ${i}",
+                       ("s",op.symbol)("p",prefix)("i", op.issuer(d).name) );
+         }
       }
-   }
-   else
-   {
-      auto dotpos = op.symbol.rfind( '.' );
-      if( dotpos != std::string::npos )
+      else
       {
-         auto prefix = op.symbol.substr( 0, dotpos );
-         auto asset_symbol_itr = asset_indx.find( prefix );
-         FC_ASSERT( asset_symbol_itr != asset_indx.end(), "Asset ${s} may only be created by issuer of ${p}, but ${p} has not been registered",
-                    ("s",op.symbol)("p",prefix) );
-         FC_ASSERT( asset_symbol_itr->issuer == op.issuer, "Asset ${s} may only be created by issuer of ${p}, ${i}",
-                    ("s",op.symbol)("p",prefix)("i", op.issuer(d).name) );
-      }
-   }
+         auto dotpos = op.symbol.rfind( '.' );
+         if( dotpos != std::string::npos )
+         {
+            auto prefix = op.symbol.substr( 0, dotpos );
+            auto asset_symbol_itr = asset_indx.find( prefix );
+            FC_ASSERT( asset_symbol_itr != asset_indx.end(), "Asset ${s} may only be created by issuer of ${p}, but ${p} has not been registered",
+                       ("s",op.symbol)("p",prefix) );
+            FC_ASSERT( asset_symbol_itr->issuer == op.issuer, "Asset ${s} may only be created by issuer of ${p}, ${i}",
+                       ("s",op.symbol)("p",prefix)("i", op.issuer(d).name) );
+         }
+
+         if(d.head_block_time() <= HARDFORK_CORE_620_TIME ) { // TODO: remove this check after hf_620
+            FC_ASSERT(isalpha(op.symbol.back()), "Asset ${s} must end with alpha character before hardfork 620", ("s",op.symbol));
+         }
+         else {
+            FC_ASSERT(isalnum(op.symbol.back()), "Asset ${s} must end with alpha or number character", ("s",op.symbol));
```

Comment:
Move this check to `validate()`.

Reply:
moved https://github.com/bitshares/bitshares-core/pull/705/commits/2f33311603dbcdb031776667140ff48770328391